src/plugins/intel_gpu/tests/transformations/decompose_reduce_for_false_keepdims_test.cpp

```diff
@@ -0,0 +1,267 @@
+    std::shared_ptr<ngraph::Function> fc, fc_ref, fc_cldnn;
+    bool need_decompose;
+
+    void SetUp() override {
+        const auto& reduce_type = std::get<0>(GetParam());
+        const auto& input_shape = std::get<1>(GetParam());
+        const auto& axes = std::get<2>(GetParam());
+        const auto& keep_dims = std::get<3>(GetParam());
+        need_decompose = std::get<4>(GetParam());
+        const auto& reshape_shape = std::get<5>(GetParam());
+
+        fc = get_transformed_function(input_shape, axes, reduce_type, keep_dims);
+        fc_ref = get_reference_function(input_shape, axes, reduce_type, reshape_shape);
+        fc_cldnn = get_transformed_function(input_shape, axes, reduce_type, false);
+    }
+
+    static std::shared_ptr<ngraph::Function> get_reference_function(const ngraph::PartialShape& input_shape,
+                                                                    const std::vector<int64_t>& axes,
+                                                                    const ReduceType& reduce_type,
+                                                                    const std::vector<size_t>& reshape_shape) {
+        auto param = std::make_shared<ngraph::opset10::Parameter>(ngraph::element::f32, input_shape);
+        if (reduce_type == reduce_mode::logical_or || reduce_type == reduce_mode::logical_and)
+            param = std::make_shared<ngraph::opset10::Parameter>(ngraph::element::boolean, input_shape);
+
+        ngraph::Output<ngraph::Node> input = param->output(0);
+        auto axes_const = ngraph::opset10::Constant::create(ngraph::element::i64, ngraph::Shape{axes.size()}, axes);
+
+        if (reduce_type == reduce_mode::sum)
+            input = std::make_shared<ngraph::opset10::ReduceSum>(input, axes_const, true);
+        else if (reduce_type == reduce_mode::mean)
+            input = std::make_shared<ngraph::opset10::ReduceMean>(input, axes_const, true);
+        else if (reduce_type == reduce_mode::min)
+            input = std::make_shared<ngraph::opset10::ReduceMin>(input, axes_const, true);
+        else if (reduce_type == reduce_mode::max)
+            input = std::make_shared<ngraph::opset10::ReduceMax>(input, axes_const, true);
+        else if (reduce_type == reduce_mode::prod)
+            input = std::make_shared<ngraph::opset10::ReduceProd>(input, axes_const, true);
+        else if (reduce_type == reduce_mode::logical_or)
+            input = std::make_shared<ngraph::opset10::ReduceLogicalOr>(input, axes_const, true);
+        else if (reduce_type == reduce_mode::logical_and)
+            input = std::make_shared<ngraph::opset10::ReduceLogicalAnd>(input, axes_const, true);
+        else
+            throw std::runtime_error("Invalid reduce type for this test-case.");
+
+        input = std::make_shared<ngraph::opset10::Reshape>(
+            input,
+            ngraph::opset10::Constant::create(ngraph::element::i64, ngraph::Shape{reshape_shape.size()}, reshape_shape),
+            false);
```

Comment:
why do we need reshape here? From previous discussion, the conclusion was to use "native" shape from ngraph, and decomposition does not change it. Then we don't need reshape here. Am I missing something?

Reply:
I will remove both fc_ref and fc_cldnn which are confusing.
It will only uses input tensor comparison from ReshapeShape.